OpenRA.Mods.Common/Traits/Mobile.cs

```diff
@@ -666,6 +681,14 @@ public override bool Tick(Actor self)
 				QueueChild(mobile.VisualMove(self, pos, self.World.Map.CenterOfSubCell(cell, subCell)));
 				return true;
 			}
+
+			public override void Cancel(Actor self, bool keepQueue = false)
+			{
+				if (!mobile.CanStayInCell(cell))
+					QueueChild(new Move(self, cell, WDist.Zero, null, true));
```

Comment:
Can we please add a comment here explaining that this is essentially nudging the actor into a valid nearby cell?

Reply:
Done.